### PR TITLE
Add stop generating button to cody web

### PR DESCRIPTION
**Pull request by thenamankumar**
Add stop generating button for in progress message for cody in web.
image

https://www.loom.com/share/811352d2fb6942ff9851c9dedda281e1
## Test plan

- visit /cody
- ask a question
- click on stop generating button
- reload the page

### CODE REVIEW

**Review by almeidapaulooliveira — APPROVED** [review]

File: client/web/src/cody/components/ChatUI/ChatUi.tsx [hunk 1]
```diff
@@ -102,11 +103,25 @@ export const ChatUI: React.FC<IChatUIProps> = ({ codyChatStore }): JSX.Element =
...
+    function AbortMessageInProgressButton({ onAbortMessageInProgress }) {
+        return (
+            <div className="d-flex justify-content-center w-100 mt-4 mb-2">
+                <Button onClick={onAbortMessageInProgress} variant="secondary" outline={true} size="sm">
+                    Stop generating
+                </Button>
+            </div>
+        )
+    }
+)
```
Comment (valerybugakov) [inline client/web/src/cody/components/ChatUI/ChatUi.tsx:123]: A couple of suggestions:
1. Extract props into a standalone interface to be consistent with other React components in the monorepo.
2. Do not use `React.memo` [if it's not required to fix a known performance issue](https://kentcdodds.com/blog/usememo-and-usecallback).

```suggestion
interface AbortMessageInProgressProps {
  onAbortMessageInProgress: () => void
}

const AbortMessageInProgress: React.FC<AbortMessageInProgressProps> = props => (
            <div className="d-flex justify-content-center w-100 mt-4 mb-2">
                <Button onClick={props.onAbortMessageInProgress} variant="secondary" outline={true} size="sm">
                    Stop generating
                </Button>
            </div>
        )
    }
)
```

File: client/cody-ui/src/chat/TranscriptItem.tsx [hunk 2]
```diff
@@ -74,6 +76,8 @@ export const TranscriptItem: React.FunctionComponent<
     copyButtonOnSubmit,
     submitButtonComponent: SubmitButton,
     chatInputClassName,
+    abortMessageInProgressComponent: AbortMessageInProgressButton,
+    onAbortMessageInProgress = () => {},
```
Comment (valerybugakov) [inline client/cody-ui/src/chat/TranscriptItem.tsx:80]: This creates a new function reference on every render, which makes the `React.memo` wrapper on the `AbortMessageInProgressButton` redundant.

**Review by valerybugakov — APPROVED** [review]

**Comment by philipp-spiess** [comment]
Omg it's beautiful @thenamankumar. Where do I bribe you so that we can also add this to VS Code? 👀

**Comment by eseliger** [comment]
@thenamankumar what's required to get this merged?

**Pull request merged**